backend/chainlit/server.py

```diff
@@ -217,7 +219,17 @@ def get_build_dir(local_target: str, packaged_target: str) -> str:
     allow_headers=["*"],
 )
 
-app.add_middleware(GZipMiddleware)
+
+class SafariWebSocketsCompatibleGZipMiddleware(GZipMiddleware):
+    async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
+        # Prevent gzip compression for HTTP requests to socket.io path due to a bug in Safari
+        if SOCKET_IO_PATH in scope["path"]:
```

Comment:
You'll need to use `scope.get("path", "")` to make this reliably work. Scope doesn't always have "path".

Reply:
@jdb78 I missed `scope["type"] == "http"` check. Also found [this gem](https://www.starlette.io/middleware/#sending-eager-responses) in depths of Starlette documentation